Configuration_adv.hpp

```diff
@@ -348,12 +408,188 @@
+  // Altitude configuration
+  #if (ALT_STEPPER_TYPE == STEPPER_TYPE_28BYJ48) && (ALT_DRIVER_TYPE == DRIVER_TYPE_ULN2003)
+    // Valid ALT stepper and driver combination
+  #elif (ALT_STEPPER_TYPE == STEPPER_TYPE_NEMA17) && (ALT_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART)
+    // Valid ALT stepper and driver combination
+  #else
+    #error Unsupported configuration. Use at own risk.
+  #endif
+#else
+  #error Unsupported configuration. Use at own risk.
+#endif 
+
+// Interfaces
+#if !defined(BLUETOOTH_ENABLED)
+  // Baseline configuration without Bluetooth is valid
+#elif defined(ESP32)
+  // Bluetooth is only supported on ESP32
+#else
+  #error Unsupported configuration. Use at own risk.
+#endif
+
+#if !defined(WIFI_ENABLED)
+  // Baseline configuration without WiFi is valid
+#elif defined(ESP32)
+  // Wifi is only supported on ESP32
+#else
+  #error Unsupported configuration. Use at own risk.
+#endif
+
+// External sensors
+#if (USE_GPS == 0)
+  // Baseline configuration without GPS is valid
+#elif defined(__AVR_ATmega2560__)
+  // GPS is only supported on ATmega
+#else
+  #error Unsupported configuration. Use at own risk.
+#endif
+
+#if (USE_GYRO_LEVEL == 0)
+  // Baseline configuration without gyro is valid
+#elif defined(ESP32) || defined(__AVR_ATmega2560__)
+  // GPS is supported on ESP32 and ATmega
+#else
+  #error Unsupported configuration. Use at own risk.
+#endif
 
-#if DEC_HOLDCURRENT < 1 || DEC_HOLDCURRENT > 31
-#error "Holdcurrent has to be within 1 and 31!"
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//                            ////////
+// VALIDATE PIN ASSIGNMENTS   ////////
+//                            ////////
+//////////////////////////////////////
+
+// Motor & driver configurations
+#if (DEC_DRIVER_TYPE == DRIVER_TYPE_ULN2003)
+  #if !defined(DEC_IN1_PIN) || !defined(DEC_IN2_PIN) || !defined(DEC_IN3_PIN) || !defined(DEC_IN4_PIN)
+     // Required pin assignments missing
+  #endif
+#elif (DEC_DRIVER_TYPE == DRIVER_TYPE_A4988_GENERIC) || (DEC_DRIVER_TYPE == DRIVER_TYPE_TMC2209_STANDALONE)
+  #if !defined(DEC_STEP_PIN) || !defined(DEC_DIR_PIN) || !defined(DEC_EN_PIN) || !defined(DEC_MS0_PIN) || !defined(DEC_MS1_PIN) || !defined(DEC_MS2_PIN)
+     // Required pin assignments missing
+  #endif
+#elif (DEC_DRIVER_TYPE == DRIVER_TYPE_TMC2209_UART)
+  #if !defined(DEC_STEP_PIN) || !defined(DEC_DIR_PIN) || !defined(DEC_EN_PIN) || !defined(DEC_DIAG_PIN) || !defined(DEC_SERIAL_PORT_TX) || !defined(DEC_SERIAL_PORT_RX)
+     // Required pin assignments missing
+  #endif
 #endif
-#if RA_RMSCURRENT > 2000 || DEC_RMSCURRENT > 2000
-#error "Do you really want to set the RMS motorcurrent above 2 Ampere? Thats almost 3A peak! Delete this error if you know what youre doing" 
+
+#if (RA_DRIVER_TYPE == DRIVER_TYPE_ULN2003)
+  #if !defined(RA_IN1_PIN) || !defined(RA_IN2_PIN) || !defined(RA_IN3_PIN) || !defined(RA_IN4_PIN)
+     // Required pin assignments missing
```

Comment:
Don't we want to generate a #error here?

Reply:
Yes. I have added errors to all these. Somehow it missed the commit but it is incoming.

Reply:
Done. I have moved the #include of Configuration_pins into Configuration_adv so they are always picked up by the validation code. I think it would make sense to combine Configuration_pins and Configuration_local_xxx as the pin assignments are often specific to a physical hardware build (I've had to recreate Configuration_pins a couple of times after it got overwritten by upstream changes). Then the configuration construction order would be Configuration_local_xxx > Configuration > Configuration_adv, each able override the next. Let me know if you're happy for me to do this.

Reply:
Also, I saw on reddit that people were having trouble with storing of roll offsets. I have confirmed that this works correctly (at least in this PR).

Reply:
Mh ... i somehow miss the commit you mentioned with added `#error`